File: Java/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map <String, Integer> map = new HashMap<>();
        
        // 참가자 카운트
        for (String name : participant) {
            map.put(name, map.getOrDefault(name,0) + 1);
        }
        
        // 완주자 카운트 감소
        for (String name: completion) {
            map.put(name, map.get(name) -1);
        }
        
        // value가 0 이상인 사람(key) 반환
        for (Map.Entry<String, Integer> entry : map.entrySet()) {
            if (entry.getValue() > 0) {
                return entry.getKey();
            }
        }
        return "";
    }
}

File: Python3/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    
    participants_dict = {}
    
    # 이름별 개수 세기(동명이인 처리)
    for person in participant:
        participants_dict[person] = participants_dict.get(person, 0) + 1
    
    # 완주자들은 참가자 count에서 1명씩 빼서 제거
    for complete_person in completion:
        participants_dict[complete_person] -=1
    
    # 완주자들은 count가 0이기 때문에 0이상인 사람만 return
    for person, count in participants_dict.items():
        if count > 0:
            return person